# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.h
# Commit: 5dfb9fc15fba
# Full Hash: 5dfb9fc15fba710d15d5720e03f6bb398d7b6fbd
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 16:09:44
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 4: Untie frame iteration from Frame::tls. r=lth
#   
#   
#   Here we replace usage of Frame::tls in frame iteration with GetNearestEffectiveTls.
#   We also maintain current tls for frame iteration object to not to call GetNearestEffectiveTls everytime.
# ==============================================================================

diff -r 3cfa16c9f9a9 -r 5dfb9fc15fba js/src/wasm/WasmFrameIter.h
--- a/js/src/wasm/WasmFrameIter.h	Fri Oct 23 18:17:37 2020 +0000
+++ b/js/src/wasm/WasmFrameIter.h	Fri Oct 23 18:17:37 2020 +0000
@@ -65,6 +65,7 @@
   const CodeRange* codeRange_;
   unsigned lineOrBytecode_;
   Frame* fp_;
+  TlsData* tls_;
   uint8_t* unwoundIonCallerFP_;
   jit::FrameType unwoundIonFrameType_;
   Unwind unwind_;
@@ -95,6 +96,7 @@
   jit::FrameType unwoundIonFrameType() const;
   uint8_t* unwoundIonCallerFP() const { return unwoundIonCallerFP_; }
   Frame* frame() const { return fp_; }
+  TlsData* tls() const { return tls_; }
 
   // Returns the address of the next instruction that will execute in this
   // frame, once control returns to this frame.
